include/kframework/optimizations.md

```diff
@@ -9,10 +9,10 @@ requires "evm.md"
 module EVM-OPTIMIZATIONS-LEMMAS [kore, symbolic]
     imports EVM
 
-    rule #sizeWordStack(WS           , N) => #sizeWordStack(WS, 0) +Int N requires N =/=Int 0                [simplification]
-    rule #sizeWordStack(WS [ I := _ ], N) => #sizeWordStack(WS, N)        requires I <Int #sizeWordStack(WS) [simplification]
-
-    rule #stackUnderflow(WS, N) => false requires N <=Int #sizeWordStack(WS) [simplification]
+    rule #sizeWordStack(WS           , N)               => #sizeWordStack(WS, 0) +Int N requires N =/=Int 0                [simplification]
+    rule #sizeWordStack(WS [ I := _ ], N)               => #sizeWordStack(WS, N)        requires I <Int #sizeWordStack(WS) [simplification]
+    rule #sizeWordStack(_            , 0)        <Int N => false                        requires N <=Int 0                 [simplification]
+    rule #sizeWordStack(_            , 0) +Int N <Int N => false                                                           [simplification]
```

Comment:
Why are these rules added here in particular? Are they needed for verying the optimizations, or just needed in general?

Also, I'm not sure why the last rule is needed, can we instead add a rule to `include/kframework/lemmas/int-simplification.k` which says `X +Int Y <Int Y => false requires 0 <=Int X`, or a rule which says `X +Int Y <Int Z => X <Int Z -Int Y [concrete(Y, Z), symbolic(X)]`, for example?